resources/metadata/hadolint.yaml

```diff
@@ -0,0 +1,64 @@
+metadata:
+  name: hadolintExecuteScan
+  description: Executes the Haskell Dockerfile Linter which is a smarter Dockerfile linter that helps you build [best practice](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/) Docker images.
+  longDescription: |-
+    Executes the Haskell Dockerfile Linter which is a smarter Dockerfile linter that helps you build [best practice](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/) Docker images.
+    The linter is parsing the Dockerfile into an abstract syntax tree (AST) and performs rules on top of the AST.
+spec:
+  inputs:
+    params:
+    - name: configurationUrl
+      type: string
+      description: URL pointing to the .hadolint.yaml exclude configuration to be used for linting. Also have a look at `configurationFile` which could avoid central configuration download in case the file is part of your repository.
+      mandatory: false
+      scope:
+      - PARAMETERS
+      - STAGES
+      - STEPS
+      default: null
+    - name: dockerFile
+      type: string
+      description: Dockerfile to be used for the assessment.
+      mandatory: false
+      scope:
+      - GENERAL
+      - PARAMETERS
+      - STAGES
+      - STEPS
+      default: ./Dockerfile
+    - name: configurationFile
+      type: string
+      description: Name of the configuration file used locally within the step. If a file with this name is detected as part of your repo downloading the central configuration via `configurationUrl` will be skipped. If you change the file's name make sure your stashing configuration also reflects this.
+      mandatory: false
+      scope:
+      - PARAMETERS
+      - STAGES
+      - STEPS
+      default: .hadolint.yaml
+      # only Groovy
+    - name: qualityGates
```

Comment:
@OliverNocon How do I mark parameters as "only relevant for groovy"?

Reply:
There are no parameters "only relevant for groovy" ...

The step parameters are for the go part and the context parameters are retrieved from the CI/CD system (i.e. Jenkins).

Maybe we have a quick chat about this and then document the outcome here ...